docs/index.adoc

```diff
@@ -70,8 +70,72 @@ owner of `TBTCVault` can initiate and finalize an upgrade to another vault.
 The new vault receives ownership of TBTC token as well as the entire `TBTCVault`
 balance in the Bank.
 
+=== Bridge, Bank, Vault Separation
+
+The system is decomposed into three separate pieces to separate concerns. The
+<<bridge,Bridge>> is concerned with being a decentralized Bitcoin custody, and
+providing a way to let the <<bank,Bank>> know whenever Bitcoin has successfully
+crossed from Bitcoin to another chain or visa versa.
+
+The <<bank,Bank>> is concerned with keeping track of Bitcoin account balances on
+Ethereum. It tracks when folks send those balances to each other, and when
+folks lock up their balances in <<vault,Vaults>>.
+
+Vaults are authorized smart contracts that can custody a balance and then do
+_anything_. For example, TBTC (the token) is implemented as our flagship Vault.
+You let it custody your Bank balance and it gives you TBTC tokens. You give it
+back TBTC tokens and it gives you back your Bank balance.
+
+[bridge]
+==== Bridge
+
+The Bridge is our interface to the Bitcoin blockchain and the off-chain
+clients. It is the Bridge's responsiblity to listen to <<depositing,deposit
+reveals>>, <<sweeping,sweep>> them, prove it did it, and then report any relevant
+information to the <<bank,Bank>>.
+
+When it comes time to redeem, it is the Bridge's responsiblity to listen to
+<<redeeming,redemption requests>>, disperse the funds, prove it did it, and then
+report any relevant information to the <<bank,Bank>>.
+
+Finally, everything about Bitcoin <<wallet-lifecycle,wallets>> falls in the
+purview of the Bridge.
+
+[bank]
+==== Bank
+
+The Bank provides Bitcoin balance accounting for Ethereum addresses with two mappings:
+
+```
+mapping(address => uint256) public balanceOf;
+mapping(address => mapping(address => uint256)) public allowance;
+```
+
+Notably, the Bank does not issue an ERC-20 token, it just does accounting. The
+`balanceOf` mapping keeps track of an Ethereum address's Bitcoin balance (in
+satoshis). Meanwhile, the `allowance` mapping keeps track of how many satoshis
+a _spender_ may transfer on an Ethereum address's behalf. Each Ethereum address
+may have many spenders, each with different allowances.
+
+A notable use of an `allowance` is to authorize a <<vault,Vault>> to custody your
+funds, but allowances are more general.
+
+[vault]
+==== Vault
+
+A Vault is an authorized Contract that interfaces with the <<bank,Bank>>. These
+are flexible, and allow us (and third parties) to write logic on top of the
+Bank architecture to implement financial systems built from Bitcoin balances.
+
+For example, TBTC is implemented as a Vault. A user with a Bank balance gives
+the `TBTCVault` an `allowance`, and then it uses that allowance to transfer
+itself Bank Balance to hold on behalf of the user, in exchange for minting the
+user the equivalent amount of TBTC tokens. Later, anyone with TBTC tokens can
+send them back to the `TBTCVault` and the vault will grant them Bank balance.
+
 == The Mechanism
```

Comment:
We have a few more typoes in this file that we could fix:
TransaprentUpgradeableProxy -> TransparentUpgradeableProxy
Transfering -> Transferring

Reply:
[4db732c](https://github.com/keep-network/tbtc-v2/pull/344/commits/4db732c4ca25b0c6c83f45877f6e36df10baa25a)